### PR TITLE
Add unit tests for BIND, CGROUP_SOCK_ADDR, and SOCK_OPS

### LINKED ISSUE #1869: Missing unit tests for BIND, CGROUP_SOCK_ADDR, and SOCK_OPS
### Describe the feature you'd like supported

netebpfext_unit supports XDP today but not the other three prog types exposed by netebpfext.

### Proposed solution

Add tests for the other three that work similar to the existing XDP unit test.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dthaler**
## Description

* Move functionality common between netebpfext_unit.cpp and netebpfext_fuzzer.cpp into netebpf_ext_helper.cpp
* Add unit tests for BIND, CGROUP_SOCK_ADDR, and SOCK_OPS

Fixes #1869 

## Testing

This PR adds tests.

## Documentation

No impact.

### CODE REVIEW

File: netebpfext/user/fwp_um.cpp [hunk 1]
```diff
@@ -68,6 +70,203 @@ _fwp_engine::classify_test_packet(_In_ const GUID* layer_guid, NET_IFINDEX if_in
     return result.actionType;
 }
 
+const uint32_t _test_destination_ipv4_address = 0x01020304;
```
Comment (dv-msft) [inline netebpfext/user/fwp_um.cpp:73]: Generally speaking, IMO we should use the ```constexpr``` qualifier for compile time constants.

File: netebpfext/user/fwp_um.cpp [hunk 2]
```diff
@@ -68,6 +70,203 @@ _fwp_engine::classify_test_packet(_In_ const GUID* layer_guid, NET_IFINDEX if_in
...
+    std::unique_lock l(lock);
+
+    FWPS_INCOMING_VALUE0 incoming_value[FWPS_FIELD_ALE_RESOURCE_ASSIGNMENT_V4_MAX] = {};
+    incoming_value[FWPS_FIELD_ALE_RESOURCE_ASSIGNMENT_V4_IP_LOCAL_PORT].value.uint16 = _test_destination_port;
+    incoming_value[FWPS_FIELD_ALE_RESOURCE_ASSIGNMENT_V4_IP_LOCAL_ADDRESS].value.uint32 =
+        _test_destination_ipv4_address;
+    incoming_value[FWPS_FIELD_ALE_RESOURCE_ASSIGNMENT_V4_IP_PROTOCOL].value.uint8 = _test_protocol;
+    incoming_value[FWPS_FIELD_ALE_RESOURCE_ASSIGNMENT_V4_ALE_APP_ID].value.byteBlob = &_test_app_id;
+
+    return test_callout(FWPS_LAYER_ALE_RESOURCE_ASSIGNMENT_V4, FWPM_LAYER_ALE_RESOURCE_ASSIGNMENT_V4, incoming_value);
```
Comment (dv-msft) [inline netebpfext/user/fwp_um.cpp:97]: we're calling ```test_callout``` with the the ```_fwp_engine::lock``` held (above).  This function seems to do a _lot_ of possibly un-bounded work (call other functions) and I was wondering if we needed the lock for all that time.  Was also curious about the potential for an (indirect) deadlock in some subsequent code-path(s).

Comment (dthaler) [reply netebpfext/user/fwp_um.cpp:97]: that was already the precedent in this file (see line 15) so is not changed by this PR.  If it is an issue, it was already an issue and should be tracked by a different bug.

Comment (dv-msft) [reply netebpfext/user/fwp_um.cpp:97]: Filed an [issue](https://github.com/microsoft/ebpf-for-windows/issues/1898) to address this.

File: netebpfext/user/fwp_um.cpp (hunk 1 above)
Comment (dthaler) [reply netebpfext/user/fwp_um.cpp]: Done

**Review by dv-msft — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Pull request merged**